src/nvim/ops.c

```diff
@@ -2602,8 +2602,8 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
 
 static void yank_copy_line(yankreg_T *reg, struct block_def *bd, size_t y_idx)
 {
-  char_u *pnew = xmallocz((size_t)(bd->startspaces + bd->endspaces
-                                   + bd->textlen));
+  char_u *pnew = xmallocz((size_t)bd->startspaces + (size_t)bd->endspaces
+                          + (size_t)bd->textlen);
```

Comment:
In general I think it's worth adding `assert(foo > 0` whenever we cast to `size_t`, unless it's a local variable. Otherwise we are just making assumptions that can break "at a distance".

Reply:
ack. will fix

Reply:
fxied